crates/ln-dlc-node/src/node.rs

```diff
@@ -112,17 +112,70 @@ impl Display for NodeInfo {
 const MILLIONTH_ROUTING_FEE: u32 = 20000;
 
 impl Node {
+    /// Constructs a new node to be run as the app
+    pub async fn new_app(
+        alias: String,
+        network: Network,
+        data_dir: &Path,
+        address: SocketAddr,
+        electrs_origin: String,
+        seed: Bip39Seed,
+        ephemeral_randomness: [u8; 32],
+    ) -> Self {
+        let user_config = app_config();
+        Node::new(
+            alias,
+            network,
+            data_dir,
+            address,
+            electrs_origin,
+            seed,
+            ephemeral_randomness,
+            user_config,
+        )
+        .await
+    }
+
+    /// Constructs a new node to be run for the coordinator
+    ///
+    /// The main difference between this and `new_app` is that the user config is different to
+    /// be able to create just-in-time channels and 0-conf channels towards our peers.
+    pub async fn new_coordinator(
+        alias: String,
+        network: Network,
+        data_dir: &Path,
+        address: SocketAddr,
+        electrs_origin: String,
+        seed: Bip39Seed,
+        ephemeral_randomness: [u8; 32],
+    ) -> Self {
+        let user_config = coordinator_config();
+        Self::new(
+            alias,
+            network,
+            data_dir,
+            address,
+            electrs_origin,
+            seed,
+            ephemeral_randomness,
+            user_config,
+        )
+        .await
+    }
```

Comment:
I suppose we'll eventually move this code to the specific crates, but this is a good first step. Thank you!